wgpu-types/src/lib.rs

```diff
@@ -451,6 +447,21 @@ bitflags::bitflags! {
     }
 }
 
+bitflags::bitflags! {
+    #[derive(Default)]
+    #[cfg_attr(feature = "trace", derive(serde::Serialize))]
+    #[cfg_attr(feature = "replay", derive(serde::Deserialize))]
+    pub struct ShaderFlags: u32 {
+        /// If enabled, `wgpu` will parse the shader with `Naga`
+        /// and validate it both internally and with regards to pipeline interface.
+        const VALIDATION = 1;
+        /// If enabled, `wgpu` will attempt to operate on `Naga` representation
+        /// of the shader module for both validation and translation into the
+        /// backend shader languages, where `gfx-hal` supports this.
```

Comment:
```suggestion
        /// If enabled, `wgpu` will attempt to operate on `Naga`'s internal representation
        /// of the shader module for both validation and translation into the
        /// backend shader language, on backends where `gfx-hal` supports this.
```

Reply:
I'm hesitating to call it "internal" any more, since it's in Naga public API